DungeonSource/Hero.java

```diff
@@ -115,11 +115,49 @@ public void subtractHitPoints(int hitPoints)
 	public void battleChoices(DungeonCharacter opponent, Scanner kb)
 	{
 	    numOfAttacks = attackSpeed/opponent.getAttackSpeed();
-
+	    
 		if (numOfAttacks == 0)
 			numOfAttacks++;
 
 		System.out.println("Number of turns this round is: " + numOfAttacks);
+		
+		do
+		{
+		    int choice = -1;
```

Comment:
I moved choice variable to inside of do/while loop to fix a bug where the same input choice would be used for multiple attacks. Now it will properly prompt the user to choose an attack type each time per round.